### DISCUSSION TITLE
Duplicate Middleware Applied to Route

### DISCUSSION THREAD

**Discussion by ManojKiranA** [laravel/framework#35186] [Q&A] [answered]
- Laravel Version: 8.14.0
- PHP Version: 7.4.11
- Database Driver & Version: Not neded

### Description:

When i try to apply the same middleware into group and for the seperate route inside that group middleware is getting Duplicated

### Steps To Reproduce:

1. Install Laravel Debugbar.
2. Paste the following code into `web.php` file

```
Route::middleware(['web'])
    ->group(function(){
        Route::get('/test', function () {
            return 'Hello World';
        })->middleware(['web']);
    });

```
3. Visit the route and see the **Route** tab of `Laravel Debugbar` and see the Middleware Column

![image](https://user-images.githubusercontent.com/30294553/98812296-3b88d300-2448-11eb-90af-e007772d741b.png)

**Comment by khalilst** [comment, marked as answer]
@ManojKiranA 
I couldn't confirm after testing your example.
Maybe it is an issue with Laravel Debugger and not Laravel itself!

However, this is the test result:

```console
php artisan route:list

+--------+----------+----------+------+---------+------------+
| Domain | Method   | URI      | Name | Action  | Middleware |
+--------+----------+----------+------+---------+------------+
|        | GET|HEAD | /        |      | Closure | web        |
|        | GET|HEAD | api/user |      | Closure | api        |
|        |          |          |      |         | auth:api   |
|        | GET|HEAD | test     |      | Closure | web        |
+--------+----------+----------+------+---------+------------+
```
Only single middleware applied here to `test` route.

Second test was the following line in `StartSession.php@handle()` from the `web` middleware group:
```php
echo (__file__ . '<br/>');
```
And this is the result from the above test:
```http
/var/www/test/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php
Hello World
```
The filename printed only once and not duplicated.

**Comment by ManojKiranA** [comment]
Yeah i can confirm that. I will open a issue in Laravel Debugbar
